executors/notifiersHandler.go

```diff
@@ -0,0 +1,110 @@
+package executors
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/multiversx/mx-chain-core-go/core/check"
+	"github.com/multiversx/mx-chain-keys-monitor-go/core"
+)
+
+const minTimeBetweenRetries = time.Millisecond * 10
+
+type notifiersHandler struct {
+	notifiers          []OutputNotifier
+	numRetries         uint32
+	timeBetweenRetries time.Duration
+}
+
+// ArgsNotifiersHandler defines the DTO struct for the NewNotifiersHandler constructor function
+type ArgsNotifiersHandler struct {
+	Notifiers          []OutputNotifier
+	NumRetries         uint32
+	TimeBetweenRetries time.Duration
+}
+
+// NewNotifiersHandler creates a new instance of type notifiersHandler
+func NewNotifiersHandler(args ArgsNotifiersHandler) (*notifiersHandler, error) {
+	for index, notifier := range args.Notifiers {
+		if check.IfNil(notifier) {
+			return nil, fmt.Errorf("%w at index %d", errNilOutputNotifier, index)
+		}
+	}
+	if args.TimeBetweenRetries < minTimeBetweenRetries {
+		return nil, fmt.Errorf("%w provided: %v, minimum: %v", errInvalidTimeBetweenRetries, args.TimeBetweenRetries, minTimeBetweenRetries)
+	}
+
+	return &notifiersHandler{
+		notifiers:          args.Notifiers,
+		numRetries:         args.NumRetries,
+		timeBetweenRetries: args.TimeBetweenRetries,
+	}, nil
+}
+
+// NotifyWithRetry will try to send the messages to the inner list of notifiers. If one or more notifiers errored, it will retry
+// for the maximum number of retries provided. There is a delay between the retries, to maximize the chances of recovery
+func (handler *notifiersHandler) NotifyWithRetry(caller string, messages ...core.OutputMessage) error {
+	log.Debug("notifiersHandler.NotifyWithRetry",
+		"executor", caller, "num messages", len(messages), "num retries", handler.numRetries,
+		"num notifiers", len(handler.notifiers))
+
+	notifiersThatErrored := make(map[string]OutputNotifier)
+	handler.handleFirstSend(notifiersThatErrored, messages)
+	return handler.handleRetries(notifiersThatErrored, messages)
```

Comment:
If one notifier is permanently failing it does not block other notifications from happening :+1:

Reply:
yes, indeed, it will signal problems in the logs